internal/shardpack/shardpack.go

```
@@ -0,0 +1,361 @@
+		}
+		asideRestore = ""
+	}
+	return rollback, nil
+}
+
+// Prune removes sibling plan directories of the same target, keeping keepPlanHash.
+func (w *writer) Prune(root, scope, targetID, keepPlanHash string) error {
+	dir := filepath.Dir(Dir(root, scope, targetID, keepPlanHash))
+	entries, err := w.deps.ReadDir(dir)
+	if err != nil {
+		return nil // nothing written yet
```

golangci-lint (2.12.2)

[error] error is not nil (line 199) but it returns nil

(nilerr)